app/src/main/java/org/gdg/frisbee/android/eventseries/GdgEventListFragment.java

```diff
@@ -143,4 +109,39 @@ private void splitEventsAndAddToAdapter(Collection<? extends SimpleEvent> events
         Collections.reverse(past);
         return new Pair<>(upcoming, past);
     }
+
+    @Override
+    protected boolean onListLoadMore(int page, int totalItemsCount) {
+        App.getInstance().getGdgXHub().getChapterAllEventList(
+            mPlusId, page).
+            enqueue(new Callback<PagedList<Event>>() {
+                @Override
+                public void success(PagedList<Event> eventsPagedList) {
+                    List<Event> events = eventsPagedList.getItems();
+                    mAdapter.addAll(events);
+                    App.getInstance().getModelCache().putAsync(mCacheKey,
+                        mEvents,
+                        DateTime.now().plusHours(2),
+                        new ModelCache.CachePutListener() {
+                            @Override
+                            public void onPutIntoCache() {
+                                mAdapter.addAll(mEvents);
+                                setIsLoading(false);
+                            }
+                        });
+                }
+
+                @Override
+                public void failure(Throwable error) {
+                    onError(R.string.fetch_events_failed);
+                }
+
+                @Override
+                public void networkFailure(Throwable error) {
+                    onError(R.string.offline_alert);
+                }
+            }
+        );
+        return true;
```

Comment:
Added a flag mReachedEndOfList, that is only true when the return data from the API returns an empty list OR if the list returned is smaller than the number of items we requested. So the next time we won't load again.